js/modules/k6/k6.go

```diff
@@ -59,7 +59,17 @@ func (*K6) Group(ctx context.Context, name string, fn goja.Callable) (goja.Value
 	state.Group = g
 	defer func() { state.Group = old }()
 
-	return fn(goja.Undefined())
+	tags := make(map[string]string)
+	tags["group"] = g.Path
+	// Call group function and generate group_duration sample
+	startTime := time.Now()
+	ret, err := fn(goja.Undefined())
+	t := time.Now()
+	duration := t.Sub(startTime).Seconds() * 1000
```

Comment:
No need for this, use `stats.D(t.Sub(startTime))`, and also do it inline when creating the sample. You can split the sample up over multiple lines.